grudge/models/advection.py

```diff
@@ -84,8 +84,8 @@ def __init__(self, dcoll, v, inflow_u=None, flux_type="central"):
     def weak_flux(self, u_tpair):
         return advection_weak_flux(self.dcoll, self.flux_type, u_tpair, self.v)
 
-    def max_eigenvalue(self, t=None, fields=None, discr=None):
-        return np.linalg.norm(self.v)
+    def max_characteristic_velocity(self, t=None, fields=None, dcoll=None):
+        return op.norm(self.dcoll, self.v, 2)
```

Comment:
This needs to be a rank-local norm, otherwise you're double-reducing.

Reply:
Technically all of our norms are rank local right now. Not sure what you want me to do here.

Reply:
See #113 for context. Basically, anytime we use `op.norm` is a  risk until we know whether that's supposed to be local or not.

> Technically all of our norms are rank local right now.

The "right now" is exactly where I'm having a problem. Suppose that changes, then this needs to change, too. There's no need for that churn. We should just fix the interface so that user code can tell whether they'll be using a local or distributed norm. Then we don't need to change all the existing usage sites again. That's why, IMO, #113 is fairly urgent.

Reply:
Marked to address this in a separate PR: https://github.com/inducer/grudge/pull/108/commits/f18a36970bd18dfe7e890d19813d97b9f0aed5ab